src/TorchSharp/NN/Module.cs

```diff
@@ -341,15 +354,20 @@ public virtual void eval()
                     }
                 }
 
-                public virtual void zero_grad()
+                public virtual void zero_grad(bool set_to_none = true)
                 {
-                    THSNN_Module_zero_grad(handle);
+                    THSNN_Module_zero_grad(handle, set_to_none);
                     CheckForErrors();
 
                     foreach (var (_, p) in named_parameters()) {
                         var grad = p.grad();
                         if (grad is not null) {
-                            grad.zero_();
+                            if (set_to_none) {
+                                p.set_grad(null);
+                                grad.Dispose();
```

Comment:
Is it worth first detaching from its dispose scope? I think it's a no-op, but it feels cleaner.

Reply:
Meaning, call `grad.DetachFromDisposeScope().Dispose()`?

Reply:
Yes

Reply:
Done